pkg/apis/workspaces/v1alpha2/commands.go

```diff
@@ -58,6 +58,8 @@ type Command struct {
 	// Mandatory identifier that allows referencing
 	// this command in composite commands, from
 	// a parent, or in events.
+	// +kubebuilder:validation:Pattern=^([a-z0-9]([-a-z0-9]*[a-z0-9]))?$
```

Comment:
I think this pattern should be 
```suggestion
	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
```
the only difference is that it the empty string invalid. This also would match the regex used in k8s for [DNS labels](https://github.com/kubernetes/kubernetes/blob/53a9d106c4aabcd550cc32ae4e8004f32fb0ae7b/pkg/util/validation.go#L36)

Reply:
`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$` allows to create resource name with all numbers. 
I manually tried on openshift 4.1 on aws, creating an ingress/service with a numeric name would fail.
So I restrict the rule to be `^[a-z]([-a-z0-9]*[a-z0-9])?$`

Reply:
Makes sense -- good catch!